ietf-mpls-te-types.yang

```diff
@@ -0,0 +1,207 @@
+  }
+
+  identity mef-10-bwp {
+    base bandwidth-profile-type;
+    description
+      "MEF 10 Bandwidth Profile";
+  }
+
+  identity rfc-2697-bwp {
+    base bandwidth-profile-type;
+    description
+      "RFC 2697 Bandwidth Profile";
+  }
+
+  identity rfc-2698-bwp {
+    base bandwidth-profile-type;
+    description
+      "RFC 2698 Bandwidth Profile";
+  }
+
+  identity rfc-4115-bwp {
+    base bandwidth-profile-type;
+    description
+      "RFC 4115 Bandwidth Profile";
+  }
+
+/*
+ * Type Definitions (MPLS-TE)
+ */
+
+  typedef load-balancing-type {
+    description
+      "The type of load balancing used on bundled links.";
+    type enumeration {
+      enum per-flow {
+        description
+          "The load-balancing algorithm ensures that packets for
+           the same flow are forwarded on the same component link,
+           while packets for different flows can be forwarded on
+           different component links.
+
+           Packets for different flows within the same LSP can be
+           forwarded on different component links.";
+      }
+      enum per-lsps {
+        description
+          "The load-balancing algorithm ensures that packets for
+           the same LSP are forwarded on the same component link,
+           while packets for different LSPs can be forwarded on
+           different component links.
+
+           Packets for different flows within the same LSP are
+           forwarded on the same component link.";
+      }
+    }
+  }  // typedef load-balancing-type
+
+/*
+ * Groupings (to be moved to te-packet-types?)
+ */
+
+  grouping te-packet-path-bandwidth {
+    description
+      "Path bandwidth for MPLS-TP. ";
+    leaf bandwidth-profile-name{
+      type string;
+      description "Name of Bandwidth Profile.";
+    }
+    leaf bandwidth-profile-type {
+      type identityref {
+        base bandwidth-profile-type;
+      }
+      description "Type of Bandwidth Profile.";
+    }
+
+    leaf CIR {
+      type uint64;
+        description
+          "Committed Information Rate in Kbps";
+    }
+
+    leaf EIR {
+      type uint64;
+      /*
+        Need to indicate that EIR is not supported by RFC 2697
+
+        must
+            '../bw-profile-type = "etht-types:mef-10-bwp" or ' +
+            '../bw-profile-type = "etht-types:rfc-2698-bwp" or ' +
+            '../bw-profile-type = "etht-types:rfc-4115-bwp"'
+
+        must
+            '../bw-profile-type != "etht-types:rfc-2697-bwp"'
+      */
+      description
+        "Excess Information Rate in Kbps
+        In case of RFC 2698, PIR = CIR + EIR";
+    }
+
+    leaf CBS {
+      type uint64;
+      description
+        "Committed Burst Size in in KBytes";
+    }
+
+    leaf EBS {
+      type uint64;
+      description
+        "Excess Burst Size in KBytes.
+          In case of RFC 2698, PBS = CBS + EBS";
+    }
+  }
+
+  grouping te-packet-link-bandwidth {
+    description
+      "Bandwidth for MPLS-TP. ";
+    leaf mpls-tp-bandwidth {
```

Comment:
pls use the  typedef bandwidth-kbps

Reply:
See https://github.com/tsaad-dev/te/pull/104/commits/6565ce7d83432bd76208fa765f56cb991ee6b4fe